Documentation/ceph-object-bucket-notifications.md

```diff
@@ -0,0 +1,171 @@
+```yaml
+bucket-notification-<notification name>: <notification name>
+```
+The CephBucketTopic, CephBucketNotification and ObjectBucketClaim must all belong to the same namespace.
+If a bucket was created manually (not via an ObjectBucketClaim), notifications on this bucket should also be created manually. However, topics in these notifications may reference topics that were created via CephBucketTopic resources.
+
+## Topics
+A CephBucketTopic represents an endpoint (of types: Kafka, AMQP0.9.1 or HTTP), or a specific resource inside this endpoint (e.g a Kafka or an AMQP topic, or a specific URI in an HTTP server). The CephBucketTopic also holds any additional info needed for a CephObjectStore's RADOS Gateways (RGW) to connect to the endpoint. Topics don't belong to a specific bucket or notification. Notifications from multiple buckets may be sent to the same topic, and one bucket (via multiple CephBucketNotifications) may send notifications to multiple topics.
+
+## Notification Reliability and Delivery
+Notifications may be sent synchronously, as part of the operation that triggered them. In this mode, the operation is acknowledged only after the notification is sent to the topic’s configured endpoint, which means that the round trip time of the notification is added to the latency of the operation itself.
+The original triggering operation will still be considered as successful even if the notification fail with an error, cannot be delivered or times out.
+
+Notifications may also be sent asynchronously. They will be committed into persistent storage and then asynchronously sent to the topic’s configured endpoint. In this case, the only latency added to the original operation is of committing the notification to persistent storage.
+If the notification fail with an error, cannot be delivered or times out, it will be retried until successfully acknowledged.
+
+## Sample
+
+### CephBucketTopic Custom Resource
+```yaml
+apiVersion: objectbucket.io/v1alpha1
+kind: CephBucketTopic
+metadata:
+  name: my-topic [1]
+  namespace: my-app-space [2]
+spec:
+   # use one of the following endpoints
+  endpoint: http://my-notification-endpoint:8080 [3]
+  # endpoint: http://my-notification-endpoint:8080/my-topic
+  # endpoint: https://my-notification-endpoint:8443
+  # endpoint: amqp://my-rabbitmq-service:5672/vhost1
+  # endpoint: amqps://my-rabbitmq-service:5671/vhost1
+  # endpoint: kafka://my-kafka-service:9092
+  objectStoreName: my-store [4]
+  objectStoreNamespace: rook-ceph [5]
+  opaqueData: my@email.com [6]
+  persistent: false [7]
+  http: [8]
+    disableVerifySSL: true [9]
+  # amqp: [10]
+  #   disableVerifySSL: true [11]
+  #   ackLevel: broker [12]
+  #   exchange: my-exchange [13]
+  # kafka: [14]
+  #   disableVerifySSL: true [15]
+  #   ackLevel: broker [16]
+  #   useSSL: false [17]
```

Comment:
Thanks for taking care of moving the endpoint spec around. Don't forget to change the location of `endpoint` in the doc here. It's going to mean we have to re-order the stuff below also.

While we are doing that, I think it would be better to explicitly number the markdown (below) so that we can cross-reference easily using the raw text in addition to the rendered text.

I see you already changed the `design/` document, which is good. :+1:

Reply:
my mistake. forgot to change the doc (just changed the design)